### ISSUE TITLE
Cannot launch multiple pipelines simultaneously

### ISSUE THREAD

**Issue by ajlewis02** [basler/gst-plugin-pylon#25]
I have a Jetson Orin AGX Devkit (running Jetpack 5.0.2) and three acA1920-48gc cameras, connected via 10 gigabit ethernet switch. When connecting with multiple cameras sequentially, everything works as expected. When attempting to launch multiple different pipelines simultaneously, gst-launch-1.0 segfaults while attempting to set the pylonsrc element to the PAUSED state. (This happens for both pipelines, one after the other.) This behavior seems to depend on the order in which the pipelines are started - specifically, how that order relates to the alphabetical order of the camera device names.

#### Details:
My network looks like this:
```
 	(s1) -- "L3"
	 |
	 |  /-- "L2"
ORIN -- (s2) -- "L1"	
```
The pipeline I used to test this is also the minimum possible pipeline:
`gst-launch-1.0 pylonsrc device-serial-number=XXXXXXXX ! fakesink`

There are two ethernet switches, nicknamed "s1" and "s2". There is one Jetson Orin AGX Devkit, nicknamed "ORIN". There are three acA10920-48gc cameras, nicknamed "L1", "L2", and "L3". (Though these aren't the actual names of the cameras, these nicknames do preserve alphabetical order.)

When connecting two (or more!) cameras sequentially (launching one pipeline, waiting for the `pylonsrc` element to successfully hit the PAUSED state, launching the next pipeline), everything works as expected.

I ran these tests by starting one pipeline in one terminal window and immediately switching to a second terminal window and starting a second pipeline. This bug doesn't seem to require specific timing - it happens so long as two pylonsrc elements are attempting to connect to cameras at the same time, even if those elements are started on different machines (which are on the same network).

When connecting to camera L1 first and L3 second, both pipelines always crashed. When connecting to camera L3 first and camera L1 second, both pipelines always ran successfully.

When connecting to camera L1 first and L2 second, both pipelines always crashed. When connecting to camera L2 first and camera L1 second, both pipelines always ran successfully.

When connecting to camera L2 and L3, the pipeline started first always succeeded and the pipeline started second always failed. 

Might be relevant: even when both pipelines successfully ran, they only started one at a time. Also, the cameras' serial numbers are in this order when sorted numerically: L3 < L1 < L2

Thanks for taking a look at this! Let me know if you need any more details.

**Comment by michaelgruner** [comment]
Thanks for your detailed report. Let me see if I can reproduce on our end. This really caught my attention:
> , even if those elements are started on different machines (which are on the same network).

* Are you using the latest version of the element?
* Can you share the output of the following? `GST_DEBUG=2,pylon*:9 gst-launch-1.0 pylonsrc device-serial-number=XXXXXXXX ! fakesink` for the case it segfaults

**Comment by ajlewis02** [comment]
Yes, we've updated to 0.5.0 on all machines. I ran a test where I connected to cameras L1 and L2 simultaneously, with L1 first. The logs for each pipeline are attached.
[cam_L1.txt](https://github.com/basler/gst-plugin-pylon/files/10272107/cam_L1.txt)
[cam_L2.txt](https://github.com/basler/gst-plugin-pylon/files/10272108/cam_L2.txt)

Note that it only segfaulted on the pipeline for camera L1 - the pipeline for camera L2 still failed, but with an exception from the pylon library.

These logs are from the case where two pipelines are started on a single machine. If you'd like, I could grab some logs from the case where one pipeline is started on each machine.

**Comment by michaelgruner** [comment]
Thanks, I’ll take a look tomorrow. Out of curiosity, did you verify this doesn’t happen with another viewer? Pylon Viewer for example MichaelOn 20 Dec 2022, at 14:21, Alek Lewis ***@***.***> wrote:﻿
Yes, we've updated to 0.5.0 on all machines. I ran a test where I connected to cameras L1 and L2 simultaneously, with L1 first. The logs for each pipeline are attached.
cam_L1.txt
cam_L2.txt
Note that it only segfaulted on the pipeline for camera L1 - the pipeline for camera L2 still failed, but with an exception from the pylon library.
These logs are from the case where two pipelines are started on a single machine. If you'd like, I could grab some logs from the case where one pipeline is started on each machine.
